lib/hammer_cli_foreman/commands.rb

```diff
@@ -202,6 +202,9 @@ def send_request
         },
         e.resource
       )
+    rescue Exception => e
+      HammerCLIForeman.foreman_api_connection.api.clear_credentials if e.to_s.include?"401"
```

Comment:
if we would like to implement this in apipie, where should i catch the error and clear the credentials?